### PR TITLE
[Doc] Guide for adding multi-modal plugins

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR adds a new guide for developing custom multi-modal plugins. The guide is mostly a placeholder atm but will become more detailed as we support more types of multi-modal data.

### CODE REVIEW

File: vllm/multimodal/base.py [hunk 1]
```diff
@@ -43,9 +43,6 @@ def try_concat(
         *,
         device: torch.types.Device,
     ) -> BatchedTensors:
-        # Avoid initializing CUDA too early
-        import torch
-
```
Comment (ywang96) [inline vllm/multimodal/base.py:48]: Was this issue resolved?

File: docs/source/dev/multimodal/adding_multimodal_plugin.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Each modality in vLLM is represented by a :class:`~vllm.multimodal.MultiModalPlugin` and registered to :data:`~vllm.multimodal.MULTIMODAL_REGISTRY`.
+For vLLM to recognize a new modality type, you have to create a new plugin and then pass it to :meth:`~vllm.multimodal.MultiModalRegistry.register_plugin`.
+
+The remainder of this document details how to define custom :class:`~vllm.multimodal.MultiModalPlugin` s.
+
+.. note::
+  This article is a work in progress.
+
+..
```
Comment (ywang96) [inline docs/source/dev/multimodal/adding_multimodal_plugin.rst:16]: Perhaps we could take image as an example to show how to add a new modality?

**Review by ywang96 — COMMENTED** [review]
Left some comments!

File: vllm/multimodal/base.py (hunk 1 above)
Comment (DarkLight1337) [reply vllm/multimodal/base.py:48]: Yeah, it's resolved. It has nothing to do with the `torch` import itself (and you can see that I import it at the top of the module already, so this one is redundant)

File: docs/source/dev/multimodal/adding_multimodal_plugin.rst (hunk 2 above)
Comment (DarkLight1337) [reply docs/source/dev/multimodal/adding_multimodal_plugin.rst:16]: I thought you already had a plan for adding a tutorial (judging from the comment) so I haven't written up anything yet. Perhaps we can do that in another PR?

Comment (ywang96) [reply docs/source/dev/multimodal/adding_multimodal_plugin.rst:16]: Yep we can do that!

**Review by ywang96 — APPROVED** [review]

**Pull request merged**